builtin/aws/lambda/releaser.go

```diff
@@ -0,0 +1,323 @@
+package lambda
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/waypoint-plugin-sdk/component"
+	"github.com/hashicorp/waypoint-plugin-sdk/docs"
+	"github.com/hashicorp/waypoint-plugin-sdk/framework/resource"
+	sdk "github.com/hashicorp/waypoint-plugin-sdk/proto/gen"
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/builtin/aws/utils"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type Releaser struct {
+	config ReleaserConfig
+}
+
+// Config implements Configurable
+func (r *Releaser) Config() (interface{}, error) {
+	return &r.config, nil
+}
+
+// ReleaseFunc implements component.ReleaseManager
+func (r *Releaser) ReleaseFunc() interface{} {
+	return r.Release
+}
+
+// StatusFunc implements component.Status
+func (r *Releaser) StatusFunc() interface{} {
+	return r.Status
+}
+
+// DestroyFunc implements component.Destroyer
+func (r *Releaser) DestroyFunc() interface{} {
+	return r.Destroy
+}
+
+func (r *Releaser) resourceManager(log hclog.Logger) *resource.Manager {
+	return resource.NewManager(
+		resource.WithLogger(log.Named("resource_manager")),
+		resource.WithValueProvider(r.getSession),
+
+		resource.WithResource(resource.NewResource(
+			resource.WithName("function_url"),
+			// WithState is needed for `rm.Resource("function_url").State()` to succeed
+			resource.WithState(&Resource_FunctionUrl{}),
+			resource.WithCreate(r.resourceFunctionUrlCreate),
+			resource.WithDestroy(r.resourceFunctionUrlDestroy),
+		)),
+	)
+}
+
+func (r *Releaser) getSession(
+	_ context.Context,
+	log hclog.Logger,
+) (*session.Session, error) {
+	return utils.GetSession(&utils.SessionConfig{
+		Logger: log,
+	})
+}
+
+func (r *Releaser) resourceFunctionUrlCreate(
+	ctx context.Context,
+	log hclog.Logger,
+	sess *session.Session,
+	sg terminal.StepGroup,
+	ui terminal.UI,
+	dep *Deployment,
+) error {
```

Comment:
We should include StepGroup output here too. If you look at the other plugins, you'll get a good sense of the kind of wording you'd want to use.